From 4e5cf76ecaa6b20f825738b07c257f4929e4652a Mon Sep 17 00:00:00 2001
From: Mart Raudsepp <mart.raudsepp@artecdesign.ee>
Date: Fri, 16 May 2008 20:06:31 +0300
Subject: [PATCH] xf86: Add AutoConfig driver for PCI ID 1022:2081 to 'amd'

(cherry picked from commit ab9b0b36ac8ac72fc48c0abd91a83de49a18313c)
(cherry picked from commit 4fa89fbe18c929e0d36305ab47e7e17841309ffd)
... and backported to 1.4 (back to no new devprivates and "amd" driver name)
---
 hw/xfree86/common/xf86AutoConfig.c |    9 +++++++--
 1 files changed, 7 insertions(+), 2 deletions(-)

diff --git a/hw/xfree86/common/xf86AutoConfig.c b/hw/xfree86/common/xf86AutoConfig.c
index 4f6a98a..703d156 100644
--- a/hw/xfree86/common/xf86AutoConfig.c
+++ b/hw/xfree86/common/xf86AutoConfig.c
@@ -160,12 +160,17 @@ videoPtrToDriverName(pciVideoPtr info)
 {
     /*
      * things not handled yet:
-     * amd/cyrix/nsc
-     * xgi
+     * cyrix/nsc.  should be merged into geode anyway.
+     * xgi.
      */
 
     switch (info->vendor)
     {
+	case 0x1022:
+		if (info->chipType == 0x2081)
+			return "amd";
+		else
+			return NULL;
 	case 0x1142:		    return "apm";
 	case 0xedd8:		    return "ark";
 	case 0x1a03:		    return "ast";
-- 
1.5.5.1

